### ISSUE TITLE
Ref app giving error while installation

### ISSUE THREAD

**Issue by gurleen-kaur** [PredixDev/predix-rmd-ref-app#10]
Exception when running deployReferenceAppCreateDataseed.  Retrying
()
Traceback (most recent call last):
  File "scripts/installRefApp.py", line 87, in <module>
    refApp.deployReferenceAppCreateDataseed(config)
  File "/predix/predix-rmd-ref-app/scripts/refApp.py", line 288, in deployReferenceAppCreateDataseed
    deployReferenceAppCreateDataseed(config)
  File "/predix/predix-rmd-ref-app/scripts/refApp.py", line 276, in deployReferenceAppCreateDataseed
    output= uploadFileToDataseed(config.DATA_SEED_URL, data, files)
  File "/predix/predix-rmd-ref-app/scripts/refApp.py", line 247, in uploadFileToDataseed
    *encode_multipart_data (data, files))
  File "/predix/anaconda/lib/python2.7/httplib.py", line 1001, in request
    self._send_request(method, url, body, headers)
  File "/predix/anaconda/lib/python2.7/httplib.py", line 1035, in _send_request
    self.endheaders(body)
  File "/predix/anaconda/lib/python2.7/httplib.py", line 997, in endheaders
    self._send_output(message_body)
  File "/predix/anaconda/lib/python2.7/httplib.py", line 850, in _send_output
    self.send(msg)
  File "/predix/anaconda/lib/python2.7/httplib.py", line 812, in send
    self.connect()
  File "/predix/anaconda/lib/python2.7/httplib.py", line 1204, in connect
    HTTPConnection.connect(self)
  File "/predix/anaconda/lib/python2.7/httplib.py", line 793, in connect
    self.timeout, self.source_address)
  File "/predix/anaconda/lib/python2.7/socket.py", line 553, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
gaierror: [Errno -8] Servname not supported for ai_socktype
None
()
Exception when running deployReferenceAppCreateDataseed.  After repairing the problem, use "--continueFrom deployReferenceAppCreateDataseed" switch to resume the install

**Comment by gstroup** [comment]
Problem should not be happening any more, after recent updates.  Let us know if the issue persists.

**Comment by Dev79** [comment]
Hi gstroup it's still happening

**Comment by gstroup** [comment]
What OS are you running on?  Can you share a stacktrace from running the install python script?  Thanks.

**Comment by Dev79** [comment]
Hi Greg, I'm running win 8.


****************** Running pushProject for abcme_data_seed ***************
***
        Fast install =y
        retrieve jar from Artifactory
        artifactory repo=https://artifactory.predix.io/artifactory/PREDIX-EXT
        artifactory user =
        artifactory pass =
        Current Directory = C:\Users\user\predix-rmd-ref-app
        Project Directory = data-seed-service
        mvnsettings=C:\Users\user\.m2\settings.xml
        mavenRepo=
        Copying artifacts..
        Current Directory = C:\Users\user\predix-rmd-ref-app\data-seed-service
        ============================
        data-seed-service
        1.3.13
        ArtifactId derived from pom.xml = data-seed-service
        Version derived from pom.xml1.3.13
Traceback (most recent call last):
  File "C:\Users\user\predix-rmd-ref-app\scripts\refApp.py", line 266, in deploy
ReferenceAppCreateDataseed
    pushProject(config,config.dataSeedAppName, 'cf push '+config.dataSeedAppName
+' -f '+dataSeedRepoName+'/manifest.yml',dataSeedRepoName, checkIfExists="false"
)
  File "C:\Users\user\predix-rmd-ref-app\scripts\predix.py", line 298, in pushPr
oject
    getJarFromArtifactory(config, cfCommand, projectDir)
  File "C:\Users\user\predix-rmd-ref-app\scripts\predix.py", line 223, in getJar
FromArtifactory
    f = open(config.mvnsettings, 'r')
IOError: [Errno 2] No such file or directory: 'C:\\Users\\user\\.m2\\settings.xm
l'
None
()
Exception when running deployReferenceAppCreateDataseed.  Retrying
()
Traceback (most recent call last):
  File "C:\Users\user\predix-rmd-ref-app\scripts\installRefApp.py", line 58, in
<module>
    refApp.deployReferenceAppCreateDataseed(config)
  File "C:\Users\user\predix-rmd-ref-app\scripts\refApp.py", line 288, in deploy
ReferenceAppCreateDataseed
    deployReferenceAppCreateDataseed(config)
  File "C:\Users\user\predix-rmd-ref-app\scripts\refApp.py", line 266, in deploy
ReferenceAppCreateDataseed
    pushProject(config,config.dataSeedAppName, 'cf push '+config.dataSeedAppName
+' -f '+dataSeedRepoName+'/manifest.yml',dataSeedRepoName, checkIfExists="false"
)
  File "C:\Users\user\predix-rmd-ref-app\scripts\predix.py", line 298, in pushPr
oject
    getJarFromArtifactory(config, cfCommand, projectDir)
  File "C:\Users\user\predix-rmd-ref-app\scripts\predix.py", line 223, in getJar
FromArtifactory
    f = open(config.mvnsettings, 'r')
IOError: [Errno 2] No such file or directory: 'C:\\Users\\user\\.m2\\settings.xm
l'
None
()
Exception when running deployReferenceAppCreateDataseed.  After repairing the pr
oblem, use "--continueFrom deployReferenceAppCreateDataseed" switch to resume th
e install


C:\Users\user\predix-rmd-ref-app>cf

**Comment by turnerth** [comment]
The error says 
IOError: [Errno 2] No such file or directory: 'C:\Users\user\.m2\settings.xml'

That means you didn't do the set up instructions.  Your computer needs to be configured to use Maven.  And the script uses Maven to access jar files from http://artifactory.predix.com.   That Artifactory is password protected by your Predix.io username/password.  So, you have to configure Maven to know about the username/password.

**Comment by Dev79** [comment]
Hi turneth, thanks i sorted out that issue by creating a settings.xml file. However, I am still getting an error:
****************** Running deployReferenceAppCreateDataseed ******************
---
applications:
  - name: dev-predix-dataseed
    buildpack: java_buildpack
    path: target/data-seed-service-1.3.13.jar
    timeout: 120
    services:
      - developrich_rmd_asset
      - developrich_rmd_uaa
      - developrich_rmd_acs

env:
    predix_asset_name : developrich_rmd_asset
    SPRING_PROFILES_ACTIVE : cloud,clouddev
    predix_userName : rmd_admin_1
    predix_password : RMD_admin_1
    predix_oauthRestHost : 98b0f069-fd01-4673-a56d-4408632afe32.predix-uaa.run.a                                                                                            ws-usw02-pr.ice.predix.io
    predix_oauthRestPort : 80
    predix_oauthClientId : mvp3_ref_app:mvp3ref@pp
    accessTokenEndpointUrl : https://98b0f069-fd01-4673-a56d-4408632afe32.predix                                                                                            -uaa.run.aws-usw02-pr.ice.predix.io/oauth/token
    clientId : mvp3_ref_app
    clientSecret: mvp3ref@pp
    acsServiceInstanceName: developrich_rmd_acs

****************** Running pushProject for developrich_data_seed ***************                                                                                            ***
        Fast install =y
        retrieve jar from Artifactory
        artifactory repo=https://artifactory.predix.io/artifactory/PREDIX-EXT
        artifactory user =
        artifactory pass =
        Current Directory = C:\projects\predix-rmd-ref-app
        Project Directory = data-seed-service
        mvnsettings=C:\Users\user\.m2\settings.xml
        mavenRepo=
        Copying artifacts..
        Current Directory = C:\projects\predix-rmd-ref-app\data-seed-service
        ============================
        data-seed-service
        1.3.13
        ArtifactId derived from pom.xml = data-seed-service
        Version derived from pom.xml1.3.13
None
()
Exception when running deployReferenceAppCreateDataseed.  Retrying

**Comment by Dev79** [comment]
Also when i run cf services, i see that the timeseries service is listed under the Bronze plan while the rest of the services are under the Tiered plan, could this be part of the issue?

**Comment by turnerth** [comment]
Nope, Bronze plan is fine.

You didn't post enough of the log.  Can you attach it?  In the predix forum you can.

**Comment by Dev79** [comment]
Hi turneth thanks, can i paste the whole log here?

**Comment by turnerth** [comment]
i have no idea...if you can figure out with github then great.  Otherwise you can use the predix.io/forum.

**Comment by Dev79** [comment]
ok posting on the Predix forum

**Comment by Dev79** [comment]
Hi turnerth,

I have posted what seemed to be possible on the forum, please let me know if it's enough- thanks

https://forum.predix.io/questions/9407/exception-when-running-deployreferenceappcreatedat.html

**Comment by turnerth** [comment]
Need you to "attach" a file with ALL your script output.  See the paperclip on the forum.

**Comment by Dev79** [comment]
It was too big, let me try again

**Comment by Dev79** [comment]
trying to upload a zip file of the stacktrace to the forum.

**Comment by Dev79** [comment]
Hi Tom, finally managed to post it as a comment in the forum under my original post. Thanks in advance.
